### PR TITLE
Improved borders on teams profile pages.

### LINKED ISSUE #6107: [CAL-624] /teams /public profile - weird borders around event types
Appears to be a double border issue and the border radius around all of the list items is wrong.

Should match the appearance on single persons booking page like:

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-624](https://linear.app/calcom/issue/CAL-624/teams-public-profile-weird-borders-around-event-types)</sub>

### PR DESCRIPTION
**Pull request by JeroenReumkens**
## What does this PR do?

Improves borders and border radii on team profile pages.
Also improve hover state in dark mode, the badges were still in a light color. 

Fixes #6107

## Screenshot of result 
CleanShot 2022-12-21 at 11 17 19@2x

CleanShot 2022-12-21 at 11 16 46@2x

**Environment**: Staging(main branch)

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

Verify the layout on a teams page.

### CODE REVIEW

**Review by sean-brydon — APPROVED** [review]
This was on my list of NITs to pickup too :) 

LGTM

**Comment by PeerRich** [comment]
dynamic links still have double border https://cal-caiwr4w5g-cal.vercel.app/free+pro
https://user-images.githubusercontent.com/8019099/208887268-f6a5add5-d091-479c-872a-92ed4a788f07.mp4

**Comment by PeerRich** [comment]
your badge update breaks the event-types


https://user-images.githubusercontent.com/8019099/208887545-05ba30dd-124a-4be1-a959-41ef7811fc8e.mp4

File: packages/ui/components/badge/Badge.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ const badgeClassNameByVariant = {
   orange: "bg-orange-100 text-orange-800",
   success: "bg-green-100 text-green-800",
   green: "bg-green-100 text-green-800",
-  gray: "bg-gray-100 text-gray-800 dark:bg-transparent dark:text-darkgray-800 group-hover:bg-gray-200",
+  gray: "bg-gray-100 text-gray-800 dark:bg-transparent dark:text-darkgray-800 group-hover:bg-gray-200 dark:group-hover:bg-darkgray-200",
```
Comment (PeerRich) [inline packages/ui/components/badge/Badge.tsx:12]: this breaks event-types becasue we dont have dark mode here

**Review by PeerRich — CHANGES_REQUESTED** [review]

File: packages/ui/components/badge/Badge.tsx (hunk 1 above)
Comment (PeerRich) [inline packages/ui/components/badge/Badge.tsx:12]:
```suggestion
  gray: "bg-gray-100 text-gray-800 dark:bg-transparent dark:text-darkgray-800 group-hover:bg-gray-200",
```

**Review by PeerRich — APPROVED** [review]

**Comment by JeroenReumkens** [comment]
As discussed via Threads — those issues were present in an older commit, all fixed now! 🔥 Going ahead and merging this.

**Pull request merged**